test/unit/org/apache/cassandra/tools/nodetool/CompactionHistoryTest.java

```diff
@@ -0,0 +1,133 @@
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.tools.nodetool;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import org.apache.cassandra.cql3.CQLTester;
+import org.apache.cassandra.db.ColumnFamilyStore;
+import org.apache.cassandra.db.Keyspace;
+import org.apache.cassandra.db.SystemKeyspace;
+import org.apache.cassandra.db.compaction.OperationType;
+import org.apache.cassandra.tools.ToolRunner;
+import org.apache.cassandra.utils.FBUtilities;
+
+import static org.apache.cassandra.db.compaction.CompactionHistoryTabularData.COMPACTION_TYPE_PROPERTY;
+import static org.apache.cassandra.tools.ToolRunner.invokeNodetool;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(Parameterized.class)
+public class CompactionHistoryTest extends CQLTester
+{
+    @Parameterized.Parameter
+    public List<String> cmd;
+
+    @Parameterized.Parameter(1)
+    public String compactionType;
+    
+    @Parameterized.Parameter(2)
+    public int systemTableRecord;
+
+    @Parameterized.Parameters(name = "{index}: cmd={0} compactionType={1} systemTableRecord={2}")
+    public static Collection<Object[]> data()
+    {
+        List<Object[]> result = new ArrayList<>();
+        result.add(new Object[]{ Lists.newArrayList("compact"), OperationType.MAJOR_COMPACTION.type, 1 });
+        result.add(new Object[]{ Lists.newArrayList("garbagecollect"), OperationType.GARBAGE_COLLECT.type, 20 });
+        result.add(new Object[]{ Lists.newArrayList("upgradesstables", "-a" ), OperationType.UPGRADE_SSTABLES.type, 10 });
+        return result;
+    }
+    
+    @BeforeClass
+    public static void setup() throws Exception
+    {
+        requireNetwork();
+        startJMXServer();
+    }
+    
+    @Test
+    public void testCompactionProperties() throws Throwable
+    {
+        createTable("CREATE TABLE %s (id text, value text, PRIMARY KEY ((id)))");
+        ColumnFamilyStore cfs = Keyspace.open(keyspace()).getColumnFamilyStore(currentTable());
+        cfs.disableAutoCompaction();
+        // write SSTables for the specific key
+        for (int i = 0; i < 10; i++)
+        {
+            execute("INSERT INTO %s (id, value) VALUES (?, ?)", "key" + i, "value" + i);
+            flush(keyspace());
+            if(compactionType.equals(OperationType.GARBAGE_COLLECT.type))
+            {
+                execute("DELETE FROM %s WHERE id = ? ", "key" + i);
+                flush(keyspace());
+            }
+        }
+        
+        int sstableCount = compactionType.equals(OperationType.GARBAGE_COLLECT.type) ? 20 : 10;
```

Comment:
whether 10 or 20 is a parameter, isn't it?

Reply:
This will be changed to 10 only based on the above comment.